src/utils/logic-app/templateUtils.ts

```diff
@@ -0,0 +1,131 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Workflow } from "azure-arm-logic/lib/models";
+import * as jsyaml from "js-yaml";
+import { normalizeParameterName, normalizeResourceName } from "../stringUtils";
+
+export interface IGenerateBuildDefinitionOptions {
+    azureSubscription: string;
+    csmFile: string;
+    csmParametersFile: string;
+    location: string;
+    resourceGroupName: string;
+}
+
+export function generateBuildDefinition(options: IGenerateBuildDefinitionOptions): string {
+    const { azureSubscription, csmFile, csmParametersFile, location, resourceGroupName } = options;
+
+    const yaml = {
+        resources: [
+            {
+                repo: "self"
+            }
+        ],
+        pool: { // tslint:disable-line: object-literal-sort-keys
+            name: "Hosted"
+        },
+        steps: [
+            {
+                task: "AzureResourceGroupDeployment@2",
+                displayName: `Azure Deployment: Create Or Update Resource Group action on ${resourceGroupName}`, // tslint:disable-line: object-literal-sort-keys
+                inputs: {
+                    azureSubscription,
+                    csmFile,
+                    csmParametersFile,
+                    location,
+                    resourceGroupName
+                }
+            }
+        ]
+    };
+
+    return jsyaml.safeDump(yaml);
+}
+
+export function generateDeploymentTemplate(parameters: Record<string, any>, resources: any[]) {
+    return {
+        $schema: "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+        contentVersion: "1.0.0.0",
+        parameters,
+        resources,
+        variables: {}
+    };
+}
+
+export function generateDeploymentTemplateParameters(parameters: Record<string, any>) {
+    return {
+        $schema: "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
```

Comment:
Perhaps pull the strings from this file into constants?

Reply:
It's only used here for now.  If we start using it elsewhere, e.g., IA deployment templates, we can make a shared constant for it.